opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_mapping.py

```diff
@@ -185,10 +194,24 @@ def _create_monitored_resource(
 
     for mr_key, map_config in mapping.items():
         mr_value = None
+        fallback_value = None
         for otel_key in map_config.otel_keys:
             if otel_key in resource_attrs:
-                mr_value = resource_attrs[otel_key]
-                break
+                if (
+                    otel_key == ResourceAttributes.SERVICE_NAME
+                    and isinstance(resource_attrs[otel_key], str)
+                    and str(resource_attrs[otel_key]).startswith(
+                        _constants.UNKNOWN_SERVICE_PREFIX
+                    )
```

Comment:
You should be able to do just one of these. Either assign to a local var before doing the isinstance check or just rely on `str()` convert whatever it is to a string.

Reply:
done